Library/Homebrew/requirements/java_requirement.rb

```diff
@@ -45,16 +30,69 @@ def inspect
 
   def display_s
     if @version
-      if @version[-1] == "+"
-        op = ">="
-        version = @version[0, @version.length-1]
-      else
+      if exact_version?
         op = "="
-        version = @version
+      else
+        op = ">="
       end
-      "#{name} #{op} #{version}"
+      "#{name} #{op} #{version_num}"
     else
       name
     end
   end
+
+  private
+
+  def version_num
+    if exact_version?
+      @version
+    else
+      @version[0, @version.length-1]
+    end
+  end
+
+  def exact_version?
+    @version && @version[-1] != "+"
```

Comment:
`.chars.last` in that case is a bit more transparent. Could even go for `@version.to_s.chars.last != "+"`